Alignment/HIPAlignmentAlgorithm/plugins/HIPTwoBodyDecayAnalyzer.cc

```diff
@@ -0,0 +1,385 @@
+#include <iostream>
+#include <vector>
+#include <memory>
+#include <utility>
+#include <cmath>
+#include <string>
+#include "TString.h"
+#include "TLorentzVector.h"
+#include "TTree.h"
+#include "TH1F.h"
+#include "TMath.h"
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/EDAnalyzer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/ServiceRegistry/interface/Service.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "CommonTools/UtilAlgos/interface/TFileService.h"
+#include "DataFormats/VertexReco/interface/Vertex.h"
+#include "DataFormats/VertexReco/interface/VertexFwd.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+
+#include "TrackingTools/TrajectoryState/interface/TrajectoryStateOnSurface.h"
+#include "TrackingTools/GeomPropagators/interface/AnalyticalImpactPointExtrapolator.h"
+#include "TrackingTools/TransientTrack/interface/TransientTrackBuilder.h"
+#include "TrackingTools/Records/interface/TransientTrackRecord.h"
+#include "RecoVertex/KalmanVertexFit/interface/KalmanVertexFitter.h"
+#include "RecoVertex/VertexPrimitives/interface/ConvertToFromReco.h"
+
+
+class HIPTwoBodyDecayAnalyzer : public edm::EDAnalyzer {
+public:
+  
+  explicit HIPTwoBodyDecayAnalyzer(const edm::ParameterSet&);
+  ~HIPTwoBodyDecayAnalyzer();
+
+  edm::EDGetTokenT<reco::TrackCollection> alcareco_trackCollToken_;
+  edm::EDGetTokenT<reco::TrackCollection> refit1_trackCollToken_;
+  edm::EDGetTokenT<reco::TrackCollection> ctf_trackCollToken_;
+  edm::EDGetTokenT<reco::TrackCollection> final_trackCollToken_;
+
+  static void fillDescriptions(edm::ConfigurationDescriptions& descriptions);
+
+  TTree* tree;
+  std::vector<std::pair<TString, Float_t*>> floatBranches;
```

Comment:
I could, but this plugin is not really used other than small scale testing. Do we expect TString to be incompatible with CMSSW somehow? Definitely branching is made via const char* or TString, not std::string, so I would have to add .c_str() everywhere else tat calls TTree::Branch.